x-pack/plugins/observability/server/assistant_functions/get_slo_change_point.ts

```diff
@@ -0,0 +1,297 @@
+        
+        Include the observed period. For example, if the observed period is "24h" the output should be "over the past 24 hours".
+
+        Always include the 'metricOperation' as well, in plain english, before the field names. For example, if the 'metricOperation' is "avg", this is the "average" and applies to all fields.
+        Example "average field1, field2, and field3"
+
+        After presenting the user with the change detection data, add a new line and be sure to ask if they want to repeat the analysis
+        over a different time period, metric operation, or SLO.
+      `,
+      descriptionForUser: i18n.translate(
+        'xpack.observability.observabilityAiAssistant.functions.registerGetSLOList.descriptionForUser',
+        {
+          defaultMessage: `Gets the source index and changes in the source data for an SLO.
+          `,
+        }
+      ),
+      parameters,
+    },
+    async ({ arguments: args }, signal) => {
+      const {
+        'slo.name': name,
+        shortWindow,
+        longWindow,
+        field,
+        start,
+        end,
+        'kibana.alert.start': alertStart,
+        metricOperation = 'avg',
+      } = args;
+      console.log('aggregationType', metricOperation);
+      const params = {
+        kqlQuery: `slo.name: "${name}*"`,
+        sortBy: 'status',
+        sortDirection: 'desc',
+        page: '1',
+        perPage: '25',
+      };
+      console.log('start', start);
+      console.log('end', end);
+      console.log('shortWindow', shortWindow);
+      console.log('longWindow', longWindow);
+      console.log('alertStart', alertStart);
+      const slos = await sloClient.find.execute(params);
+      let absoluteStartTime: string | undefined;
+
+      if (alertStart && longWindow) {
+        absoluteStartTime = datemath
+          .parse(`now-${longWindow}`, { forceNow: new Date(alertStart) })
+          ?.toISOString();
+      }
+      const esClient = (await resources.context.core).elasticsearch.client.asCurrentUser;
+
+      const mappedItems = slos.results.map((item): SLOSourceItem => {
+        return {
+          name: item.name,
+          sourceIndex: item.indicator.params.index,
+        };
+      });
+      const index = mappedItems[0].sourceIndex;
+      let dataView: DataView;
+
+      const dataViewId = (await dataViewsClient.getIdsWithTitle()).find((id) => {
+        return id.title === index;
+      })?.id;
+      if (dataViewId) {
+        dataView = await dataViewsClient.get(dataViewId);
+      } else {
+        dataView = await dataViewsClient.create({
+          title: mappedItems[0].sourceIndex,
+        });
+      }
+
+      const metricFields = dataView.fields
```

Comment:
Find all the fields that are able to be used for change point detection and run the `change_point` aggregation for all of them.